cmd/version/cmd.go

```diff
@@ -17,77 +17,104 @@ limitations under the License.
 package version
 
 import (
+	"context"
 	"fmt"
 	"os"
 
 	"github.com/spf13/cobra"
 
 	verify "github.com/openshift/rosa/cmd/verify/rosa"
 	"github.com/openshift/rosa/pkg/info"
+	"github.com/openshift/rosa/pkg/reporter"
 	"github.com/openshift/rosa/pkg/rosa"
 )
 
-var (
-	args struct {
-		clientOnly bool
-		verbose    bool
+var args struct {
+	clientOnly bool
+	verbose    bool
+}
+
+type RosaVersionOptions struct {
+	reporter   *reporter.Object
+	verifyRosa verify.VerifyRosa
+}
+
+func NewRosaVersionOptions() (*RosaVersionOptions, error) {
+	verifyRosa, err := verify.NewVerifyRosaOptions()
+	if err != nil {
+		return nil, fmt.Errorf("failed to build rosa verify options : %v", err)
 	}
 
-	Cmd             = makeCmd()
-	delegateCommand = verify.Cmd.Run // used in testing
-)
+	rpt, err := reporter.CreateReporter()
```

Comment:
@ciaranRoche I'm actually not sure why the `reporter.Builder.Build()` method returns an `error`. It just returns a new pointer to the `Object`. I think we can drop the `error` return from the method signature which greatly cleans things up in terms of having to propagate that `error` everywhere. 

Probably worth doing that under another PR to keep the changes in this one focused. Let me take a look 👍

Reply:
Agree, lets coordinate on what would be the easiest option, and less of a rebase hell :laughing: We can follow up with the reporter change after this lands as part of the continuous improvement

Reply:
Just seen the PR, will merge that and rebase on this